pkg/services/authn/authnimpl/sync/oauth_token_sync.go

```diff
@@ -56,44 +62,44 @@ func (s *OAuthTokenSync) SyncOauthTokenHook(ctx context.Context, id *authn.Ident
 		return nil
 	}
 
-	ctxLogger := s.log.FromContext(ctx).New("userID", id.GetID())
+	userID, err := id.GetInternalID()
+	if err != nil {
+		s.log.FromContext(ctx).Error("Failed to refresh token. Invalid ID for identity", "type", id.GetIdentityType(), "err", err)
+		return nil
+	}
 
-	_, err, _ := s.singleflightGroup.Do(id.GetID(), func() (interface{}, error) {
+	ctxLogger := s.log.FromContext(ctx).New("userID", userID)
+
+	cacheKey := id.GetID()
```

Comment:
Probably a good idea to namespace the cache keys like it did before (will need to import `fmt` too)
```suggestion
	cacheKey := fmt.Sprintf("token-check-%s", id.GetID())
```